hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/table/ITTestHoodieDataSource.java

```diff
@@ -1358,7 +1358,11 @@ private List<Row> execSelectSql(TableEnvironment tEnv, String select, String sin
     TableResult tableResult = tEnv.executeSql("insert into sink " + select);
     // wait for the timeout then cancels the job
     TimeUnit.SECONDS.sleep(timeout);
-    tableResult.getJobClient().ifPresent(JobClient::cancel);
+    try {
+      tableResult.getJobClient().ifPresent(JobClient::cancel);
+    } catch (IllegalStateException e) {
+      log.info("MiniCluster has already been shut down, do nothing.");
+    }
```

Comment:
Thanks for the fix @xicm ! I have applied your patch but it does not solve the problem.

Actually the test is flaky because in recent change:

https://github.com/apache/hudi/blob/52fe1c9faeb83fe51b520e18d0c37b67ad3fcfe4/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/compact/CompactionPlanOperator.java#L112

We have tweaked the compaction scheduling strategy: when there is a pending compaction, we can not schedule another one, the test is flaky based on whether there is pending compaction when scheduling.

My initial idea is we change the test itself:
[4053.patch.zip](https://github.com/apache/hudi/files/8644508/4053.patch.zip)
Hope it helps, or maybe you have better idea :)

Reply:
Sorry, I didn't go into this test. my changes is not the root cause, let me close this pr and you push your patch.

Reply:
I found the testWriteAndReadDebeziumJson is flaky a few days ago,it resolved with my changes. But I haven't reproduced it today. It seems I haven't found the root cause.